File: packages/go/contracts/views/analytics.go
package views

import (
	"time"

	"trend-buddy/packages/go/contracts/domain"
)

type PerformanceSummary struct {
	Period      string         `json:"period"`
	TotalPNL    domain.Decimal `json:"total_pnl"`
	WinRate     domain.Decimal `json:"win_rate"`
	MaxDrawdown domain.Decimal `json:"max_drawdown"`
	TradeCount  int            `json:"trade_count"`
}

type PortfolioSummary struct {
	AccountID     domain.AccountID `json:"account_id,omitempty"`
	Equity        domain.Decimal   `json:"equity"`
	Cash          domain.Decimal   `json:"cash"`
	RealizedPNL   domain.Decimal   `json:"realized_pnl,omitempty"`
	UnrealizedPNL domain.Decimal   `json:"unrealized_pnl,omitempty"`
	OpenPositions int              `json:"open_positions"`
	Timestamp     time.Time        `json:"timestamp"`
}

File: packages/python/src/trend_buddy_shared/contracts/views/analytics.py
from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal


@dataclass
class PerformanceSummary:
    period: str
    total_pnl: Decimal
    win_rate: Decimal
    max_drawdown: Decimal
    trade_count: int


@dataclass
class PortfolioSummary:
    equity: Decimal
    cash: Decimal
    open_positions: int
    timestamp: datetime
    account_id: str = ""
    realized_pnl: Decimal = Decimal("0")
    unrealized_pnl: Decimal = Decimal("0")
